src/components/templates/Camp/components/Map.tsx

```diff
@@ -109,495 +90,156 @@ export const Map: React.FC<MapProps> = ({
-            const hasAttendance = attendances[`${venue.name}/${room.title}`];
+    <div
+      className="grid-container"
+      style={{
+        backgroundImage: `url(${venue.mapBackgroundImageUrl})`,
+        backgroundSize: "cover",
+        display: "grid",
+        gridTemplateColumns: `repeat(${templateColumns}, calc(100% / ${templateColumns}))`,
+        gridTemplateRows: `repeat(${templateRows}, 1fr)`,
+      }}
+    >
+      {columnsArray.map((_, colIndex) => (
+        <div className="seat-column" key={`column${colIndex}`}>
+          {rowsArray.map((_, rowIndex) => {
+            const column = colIndex + 1; // TODO: do these need to be here, can we zero index?
+            const row = rowIndex + 1; // TODO: do these need to be here, can we zero index?
+
+            const seatedPartygoer = partygoersBySeat?.[row]?.[column] ?? null;
+            const hasSeatedPartygoer = !!seatedPartygoer;
+
+            // TODO: our types imply that this shouldn't be able to be null, but it was..
+            const isMe = seatedPartygoer?.id === user.uid;
+
             return (
-              <div
-                className={`room position-absolute ${
-                  isUnderneathRoom && "isUnderneath"
-                }`}
-                style={{
-                  left: left + "%",
-                  top: top + "%",
-                  width: width + "%",
-                  height: height + "%",
-                }}
-                key={room.title}
-                onClick={() => {
-                  if (!IS_BURN) {
-                    openModal(room);
-                  } else {
-                    setRoomClicked((prevRoomClicked) =>
-                      prevRoomClicked === room.title ? undefined : room.title
-                    );
-                  }
-                }}
-                onMouseEnter={() => {
-                  dispatch(retainAttendance(true));
-                  setRoomHovered(room);
-                }}
-                onMouseLeave={() => {
-                  dispatch(retainAttendance(false));
-                  setRoomHovered(undefined);
-                }}
-              >
-                <div
-                  className={`camp-venue ${
-                    roomClicked === room.title ? "clicked" : ""
-                  }`}
-                >
-                  <div
-                    className={`grid-room-btn ${
-                      isUnderneathRoom && "isUnderneath"
-                    }`}
-                  >
-                    <div
-                      className="btn btn-white btn-small btn-block"
-                      onClick={(e) => onJoinRoom(e, room)}
-                    >
-                      {venue.joinButtonText ?? "Join now"}
-                    </div>
-                  </div>
-                  <div className="camp-venue-img">
-                    <img
-                      src={room.image_url}
-                      title={room.title}
-                      alt={room.title}
```

Comment:
this was refactored into `src/components/templates/Camp/components/MapRoomsOverlay.tsx`